app/components/chat_component.rb

```diff
@@ -0,0 +1,45 @@
+class ChatComponent < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :user, :conversation, :messages, :send_message_id
+
+  def initialize(user:, conversation:, messages:)
+    @user = user
+    @conversation = conversation
+    @messages = messages
+    @send_message_id = SecureRandom.uuid
+    self.class.stream_for(conversation, :handle_received)
+  end
+
+  map_motion :send_message
+  def send_message(event)
+    message = conversation.public_send(
+      ActiveAdmin::Chat.message_relation_name.pluralize
+    ).create!(
+      sender: user,
+      content: event.form_data[:message]
```

Comment:
I think it would be cool if this data could come already unwrapped, more like a controller's params. Phoenix has `handle_event(event, params, socket)` but this makes more sense for pattern matching.

Reply:
now that you mention I think motion provides `params`